modules/widgets/authors.php

```diff
@@ -0,0 +1,182 @@
+				echo '<a href="' . get_author_posts_url( $author->ID ) . '">';
+
+				if ( $instance['avatar_size'] > 1 )
+					echo ' ' . get_avatar( $author->ID, $instance['avatar_size'], '', true ) . ' ';
+
+				echo '<strong>' . esc_html( $author->display_name ) . '</strong>';
+				echo '</a>';
+			}
+			else if ( $instance['all'] ) {
+				if ( $instance['avatar_size'] > 1 )
+					echo get_avatar( $author->ID, $instance['avatar_size'], '', true ) . ' ';
+
+				echo '<strong>' . esc_html( $author->display_name ) . '</strong>';
+			}
+
+			if ( 0 == $instance['number'] ) {
+				echo '</li>';
+				continue;
+			}
+
+			// Display a short list of recent posts for this author
+
+
+			if ( $r->have_posts() ){
+				echo '<ul>';
+
+				while ( $r->have_posts() ) {
+					$r->the_post();
+					echo '<li><a href="' . get_permalink() . '">';
+
+					if ( get_the_title() )
+						echo get_the_title();
+					else
+						echo get_the_ID();
+
+					echo '</a></li>';
+				}
+
+				echo '</ul>';
+			}
+
+			echo '</li>';
+		}
+
+		echo '</ul>';
+		echo $args['after_widget'];
+
+		wp_reset_postdata();
+
+		if ( '%BEG_OF_TITLE%' != $args['before_title'] )
+			wp_cache_add( $cache_bucket, ob_get_flush(), 'widget');
+
+		/** This action is documented in modules/widgets/gravatar-profile.php */
+		do_action( 'jetpack_stats_extra', 'widget_view', 'authors' );
+	}
+
+	public function form( $instance ) {
+		$instance = wp_parse_args( $instance, array( 'title' => '', 'all' => false, 'avatar_size' => 48, 'number' => 0 ) );
+
+		?>
+		<p>
+			<label>
+				<?php _e( 'Title:', 'jetpack' ); ?>
+				<input class="widefat" name="<?php echo $this->get_field_name( 'title' ); ?>" type="text" value="<?php echo esc_attr( $instance['title'] ); ?>" />
+			</label>
+		</p>
+		<p>
+			<label>
+				<input class="checkbox" type="checkbox" <?php checked( $instance['all'] ); ?> name="<?php echo $this->get_field_name( 'all' ); ?>" />
+				<?php _e( 'Display all authors (including those who have not written any posts)', 'jetpack' ); ?>
+			</label>
+		</p>
+		<p>
+			<label>
+				<?php _e( 'Number of posts to show for each author:', 'jetpack' ); ?>
+				<input style="width: 50px; text-align: center;" name="<?php echo $this->get_field_name( 'number' ); ?>" type="text" value="<?php echo esc_attr( $instance['number'] ); ?>" />
+				<?php _e( '(at most 10)', 'jetpack' ); ?>
+			</label>
+		</p>
+		<p>
+			<label>
+				<?php _e( 'Avatar Size (px):', 'jetpack' ); ?>
+				<select name="<?php echo $this->get_field_name( 'avatar_size' ); ?>">
+					<?php foreach( array( '1' => __( 'No Avatars', 'jetpack' ), '16' => '16x16', '32' => '32x32', '48' => '48x48', '96' => '96x96', '128' => '128x128' ) as $value => $label ) { ?>
+						<option value="<?php echo esc_attr( $value ); ?>" <?php selected( $value, $instance['avatar_size'] ); ?>><?php echo esc_html( $label ); ?></option>
+					<?php } ?>
+				</select>
+			</label>
+		</p>
+		<?php
+	}
+
+	public function update( $new_instance, $old_instance ) {
```

Comment:
It'd be nice to get docblocks before each function, like we've done for new widgets recently.